app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.kt

```diff
@@ -0,0 +1,513 @@
+import android.content.DialogInterface
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.os.Environment
+import android.util.Log
+import android.view.MenuItem
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.PopupMenu
+import androidx.core.app.ActivityOptionsCompat
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import at.connyduck.calladapter.networkresult.fold
+import autodispose2.AutoDispose
+import autodispose2.androidx.lifecycle.AndroidLifecycleScopeProvider
+import com.google.android.material.snackbar.Snackbar
+import com.keylesspalace.tusky.BaseActivity
+import com.keylesspalace.tusky.BottomSheetActivity
+import com.keylesspalace.tusky.PostLookupFallbackBehavior
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.StatusListActivity.Companion.newHashtagIntent
+import com.keylesspalace.tusky.ViewMediaActivity.Companion.newIntent
+import com.keylesspalace.tusky.components.compose.ComposeActivity.Companion.startIntent
+import com.keylesspalace.tusky.components.compose.ComposeActivity.ComposeOptions
+import com.keylesspalace.tusky.components.report.ReportActivity.Companion.getIntent
+import com.keylesspalace.tusky.db.AccountEntity
+import com.keylesspalace.tusky.db.AccountManager
+import com.keylesspalace.tusky.di.Injectable
+import com.keylesspalace.tusky.entity.Attachment
+import com.keylesspalace.tusky.entity.Status
+import com.keylesspalace.tusky.interfaces.AccountSelectionListener
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.usecase.TimelineCases
+import com.keylesspalace.tusky.util.openLink
+import com.keylesspalace.tusky.util.parseAsMastodonHtml
+import com.keylesspalace.tusky.view.showMuteAccountDialog
+import com.keylesspalace.tusky.viewdata.AttachmentViewData
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import kotlinx.coroutines.launch
+import java.lang.IllegalStateException
+import java.util.LinkedHashSet
+import javax.inject.Inject
+
+/* Note from Andrew on Jan. 22, 2017: This class is a design problem for me, so I left it with an
+ * awkward name. TimelineFragment and NotificationFragment have significant overlap but the nature
+ * of that is complicated by how they're coupled with Status and Notification and the corresponding
+ * adapters. I feel like the profile pages and thread viewer, which I haven't made yet, will also
+ * overlap functionality. So, I'm momentarily leaving it and hopefully working on those will clear
+ * up what needs to be where. */
+abstract class SFragment : Fragment(), Injectable {
+    protected abstract fun removeItem(position: Int)
+    protected abstract fun onReblog(reblog: Boolean, position: Int)
+    private var bottomSheetActivity: BottomSheetActivity? = null
```

Comment:
```suggestion
    private lateinit var bottomSheetActivity: BottomSheetActivity
```

then you don't need any calls with ?